src/portable/portable.c

```diff
@@ -610,11 +609,7 @@ static int extract_image_and_extensions(
                         continue;
 
                 /* We need to keep the fd valid, to return the PortableMetadata to the caller. */
-                extension_release_fd = fd_reopen(extension_release_meta->fd, O_CLOEXEC|O_RDONLY);
-                if (extension_release_fd < 0)
-                        return extension_release_fd;
-
-                r = take_fdopen_unlocked(&extension_release_fd, "r", &f);
+                r = fdopen_independent(extension_release_meta->fd, "re", &f);
```

Comment:
yes. fdopen() doesn't care for the "e", so we didn#t use to specify this. But now we need it, because we derive the fd_reopen() integer flags from the string now, and there we definitely want to have O_CLOEXEC.